runtime/js-compute-runtime/builtins/cache-core.cpp

```diff
@@ -0,0 +1,1162 @@
+    options.metadata = metadata;
+  }
+
+  return options;
+}
+
+// The JavaScript TransactionInsertOptions parameter we are parsing should have the below interface:
+// interface TransactionInsertOptions {
+//   maxAge: number,
+//   vary?: Array<string>,
+//   initialAge?: number,
+//   staleWhileRevalidate?: number,
+//   surrogateKeys?: Array<string>,
+//   length?: number,
+//   userMetadata?: ArrayBufferView | ArrayBuffer | URLSearchParams | string,
+//   sensitive?: boolean,
+// }
+JS::Result<host_api::CacheWriteOptions> parseTransactionInsertOptions(JSContext *cx,
+                                                                      JS::HandleValue options_val) {
+  auto options_res = parseTransactionUpdateOptions(cx, options_val);
+  if (options_res.isErr()) {
+    return JS::Result<host_api::CacheWriteOptions>(JS::Error());
+  }
+  auto options = options_res.unwrap();
+  JS::RootedObject options_obj(cx, &options_val.toObject());
+
+  JS::RootedValue sensitive_val(cx);
+  if (!JS_GetProperty(cx, options_obj, "sensitive", &sensitive_val)) {
+    return JS::Result<host_api::CacheWriteOptions>(JS::Error());
+  }
+  options.sensitive = JS::ToBoolean(sensitive_val);
+
+  return options;
+}
+
+// The JavaScript TransactionInsertOptions parameter we are parsing should have the below interface:
+// interface InsertOptions extends TransactionInsertOptions {
+//   headers?: HeadersInit,
+// }
+JS::Result<host_api::CacheWriteOptions> parseInsertOptions(JSContext *cx,
+                                                           JS::HandleValue options_val) {
+  auto options_res = parseTransactionInsertOptions(cx, options_val);
+  if (options_res.isErr()) {
+    return JS::Result<host_api::CacheWriteOptions>(JS::Error());
+  }
+  auto options = options_res.unwrap();
+  JS::RootedObject options_obj(cx, &options_val.toObject());
+  JS::RootedValue headers_val(cx);
+  if (!JS_GetProperty(cx, options_obj, "headers", &headers_val)) {
+    return JS::Result<host_api::CacheWriteOptions>(JS::Error());
+  }
+  // headers property is optional
+  if (!headers_val.isUndefined()) {
+    JS::RootedObject headersInstance(cx, JS_NewObjectWithGivenProto(cx, &builtins::Headers::class_,
+                                                                    builtins::Headers::proto_obj));
+    if (!headersInstance) {
+      return JS::Result<host_api::CacheWriteOptions>(JS::Error());
+    }
+    auto headers = builtins::Headers::create(
+        cx, headersInstance, builtins::Headers::Mode::Standalone, nullptr, headers_val, true);
+    if (!headers) {
+      return JS::Result<host_api::CacheWriteOptions>(JS::Error());
+    }
+    JS::RootedValue headers_val(cx, JS::ObjectValue(*headers));
+    JS::RootedObject requestInstance(cx, Request::create_instance(cx));
+    if (!requestInstance) {
+      return JS::Result<host_api::CacheWriteOptions>(JS::Error());
+    }
+
+    JS::RootedValue input(cx, JS::StringValue(JS_NewStringCopyZ(cx, "http://example.com")));
```

Comment:
Could you add a comment about why this constant is used?

Reply:
Good point, I've now added one in [`38a8ce9` (#566)](https://github.com/fastly/js-compute-runtime/pull/566/commits/38a8ce9612e3af8081304979ee6983d5cfa679c8)